frontend/app/mocks/node.ts

```
@@ -1,21 +1,55 @@
+
+    if (!userEntity) {
+      return new HttpResponse('User Not found', { status: 404, headers: { 'Content-Type': 'text/plain' } });
+    }
+
     return HttpResponse.json({
-      firstName: 'John',
-      lastName: 'Maverick',
-      phoneNumber: '(555) 555-5555',
-      homeAddress: '123 Home Street',
-      mailingAddress: '123 Mailing Street',
-      preferredLanguage: 'fr'
-    })
+      id: userEntity.id,
+      firstName: userEntity.firstName,
+      homeAddress: userEntity.homeAddress,
+      lastName: userEntity.lastName,
+      mailingAddress: userEntity.mailingAddress,
+      phoneNumber: userEntity.phoneNumber,
+      preferredLanguage: userEntity.preferredLanguage,
+    });
   }),
 
-  http.patch('https://api.example.com/users/*', () => {
-    return new HttpResponse(null, { status: 204 });
+  http.patch('https://api.example.com/users/:id', async ({ params, request }) => {
+    const parsedUserId = userIdSchema.safeParse(params.id);
+    const userEntity = parsedUserId.success ? db.user.findFirst({ where: { id: { equals: parsedUserId.data } } }) : undefined;
+
+    if (!userEntity) {
+      return new HttpResponse('User Not found', { status: 404, headers: { 'Content-Type': 'text/plain' } });
+    }
+
+    try {
+      const { id, homeAddress, mailingAddress, phoneNumber, preferredLanguage } = userEntity;
+      const document = { homeAddress, mailingAddress, phoneNumber, preferredLanguage };
+      const patch = (await request.json()) as Operation[];
+      const patchResult = jsonpatch.applyPatch(document, patch, true);
+      db.user.update({ where: { id: { equals: id } }, data: patchResult.newDocument });
+    } catch (error) {
+      if (error instanceof JsonPatchError) {
+        return HttpResponse.text(error.message, { status: 400 });
+      }
+
+      return HttpResponse.text((error as Error).message, { status: 500 });
+    }
```

Do you really need to try/catch here? It's a mock so why not just let the error bubble up?

---

Good catch. I read the documentation and it returns a 500 in that case.

https://mswjs.io/docs/basics/mocking-responses/
> Note that any non-response exceptions will be translated to 500 Internal Server Error responses, similar to an unhandled rejection happening on server runtime.